Userland/Libraries/LibGfx/Filters/FastBoxBlurFilter.cpp

```diff
@@ -102,10 +96,10 @@ static void do_single_pass(int width, int height, size_t radius_x, size_t radius
         // Setup sliding window
         for (int i = -(int)radius_y; i <= (int)radius_y; ++i) {
             int offset = clamp(i, 0, height - 1) * width + x;
-            sum_red += intermediate_red[offset];
-            sum_green += intermediate_green[offset];
-            sum_blue += intermediate_blue[offset];
-            sum_alpha += intermediate_alpha[offset];
+            sum_red += intermediate[offset].red();
```

Comment:
Did you try keeping a reference to the intermediate color around? Does that effect the code gen? Or is the compiler able to generate equivalent code?

Reply:
I tried making this change in my own experiments. It looks nicer, but it does not make any noticeable change in performance. Seems like the compiler can figure it out (see similar reduced case https://godbolt.org/z/fohbEGe3P).

Reply:
Nice, thanks!

Reply:
Here is a complete demonstration of both working the same way using `AK::Vector` and with `operator[]()` instead of `at()`. https://serenityos.godbolt.org/z/PEP8MWbqx

Reply:
FYI it's the exact same on O1 through O3.

Reply:
Agreed. This is why I didn't make the change originally, but there was 1 request and 1 suggestion that it "looks nicer".